### PR TITLE
Allow overriding rubocop executable.

### PR DESCRIPTION
**Pull request by meunierd**
Rubocop configuration can get pretty hairy, to the point that you actually want one per-project rather than a global rubocop. The two standard ways for the ruby ecosystem to handle this is either through binstubs (eg. `bin/rubocop`) or bundler `bundle exec rubocop`.

### CODE REVIEW

File: ale_linters/ruby/rubocop.vim [hunk 1]
```diff
@@ -36,6 +37,10 @@ if !exists('g:ale_ruby_rubocop_options')
     let g:ale_ruby_rubocop_options = ''
 endif
 
+if !exists('g:ale_ruby_rubocop_executable')
+    let g:ale_ruby_rubocop_executable = 'rubocop'
+endif
+
 call ale#linter#Define('ruby', {
 \   'name': 'rubocop',
 \   'executable': 'rubocop',
```
Comment (w0rp) [inline ale_linters/ruby/rubocop.vim]: You should change this to `executable_callback` and add a function, so ALE which check if your override is executable.

File: ale_linters/ruby/rubocop.vim [hunk 2]
```diff
@@ -25,7 +25,8 @@ function! ale_linters#ruby#rubocop#Handle(buffer, lines) abort
 endfunction
 
 function! ale_linters#ruby#rubocop#GetCommand(buffer) abort
-    return 'rubocop --format emacs --force-exclusion '
+    return ale#Var(a:buffer, 'ale_ruby_rubocop_executable')
```
Comment (w0rp) [inline ale_linters/ruby/rubocop.vim]: You should apply `ale#Escape` to this value here, in case the path to the executable contains spaces, or whatever.

**Comment by meunierd** [comment]
Added tests and `executable_callback` 👍

**Comment by meunierd** [comment]
Updated.

File: ale_linters/ruby/rubocop.vim [hunk 3]
```diff
@@ -25,20 +25,37 @@ function! ale_linters#ruby#rubocop#Handle(buffer, lines) abort
...
+        let l:executable = l:executable . ' exec rubocop'
+    endif
+    return l:executable
+    \   . ' --format emacs --force-exclusion '
     \   . ale#Var(a:buffer, 'ruby_rubocop_options')
     \   . ' --stdin ' . bufname(a:buffer)
 endfunction
 
+function! ale_linters#ruby#rubocop#GetExecutable(buffer) abort
+    let l:executable = ale#Var(a:buffer, 'ruby_rubocop_executable')
```
Comment (w0rp) [inline ale_linters/ruby/rubocop.vim]: You just need to `return` here, as ALE uses `if executable(...)` internally. I'll fix this later.

**Comment by w0rp** [comment]
Cheers! :beers:

**Comment by meunierd** [comment]
🥂 It looks like flake8 does the same thing: https://github.com/w0rp/ale/blob/master/ale_linters/python/flake8.vim#L28

**Comment by w0rp** [comment]
Ah, that's slightly different. It's looking for an executable in virtualenv, and if that fails it fails back on a possible global executable.

**Pull request merged**